assets/js/modules/idea-hub/datastore/idea-state.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * `modules/idea-hub` data store: idea-state.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+import isPlainObject from 'lodash/isPlainObject';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from './constants';
+import { createFetchStore } from '../../../googlesitekit/data/create-fetch-store';
+import { actions as errorStoreActions } from '../../../googlesitekit/data/create-error-store';
+const { receiveError, clearError } = errorStoreActions;
+
+const fetchPostUpdateIdeaStateStore = createFetchStore( {
+	baseName: 'updateIdeaState',
+	controlCallback: ( { ideaState } ) => {
+		return API.set( 'modules', 'idea-hub', 'update-idea-state', { ideaState } );
+	},
+	argsToParams( ideaState ) {
+		return { ideaState };
+	},
+	validateParams( { ideaState } = {} ) {
+		invariant( isPlainObject( ideaState ), 'ideaState must be an object.' );
+		invariant( typeof ideaState.name === 'string', 'ideaState.name must be a string.' );
+	},
```

Comment:
Then in the validateParams callback, we can check props like this:

```suggestion
	validateParams( { name, saved, dismissed } = {} ) {
		invariant( typeof name === 'string' && name.length > 0, 'name must be a non empty string' );
		invariant( saved !== undefined || dimissed !== undefined, 'either saved or dimissed property must be set' );
	},
```